### PR TITLE
Fix multiple-active="stop_unexpected"

### PR DESCRIPTION
**Pull request by kgaillot**
The original implementation was only partial and unnecessarily complicated

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , can you review this when you get a chance? Thanks

File: lib/pacemaker/pcmk_sched_native.c [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ native_create_actions(pe_resource_t * rsc, pe_working_set_t * data_set)
     gboolean need_stop = FALSE;
     bool need_promote = FALSE;
     gboolean is_moving = FALSE;
-    gboolean allow_migrate = pcmk_is_set(rsc->flags, pe_rsc_allow_migrate)? TRUE : FALSE;
+    gboolean allow_migrate = FALSE;
```
Comment (clumens) [inline lib/pacemaker/pcmk_sched_native.c:1253]: I should have commented on this before, but if you turn allow_migrate into a bool, you can get rid of the whole `pcmk_is_set() ? TRUE : FALSE` thing, which looks weird in the first place.

File: lib/pacemaker/pcmk_sched_native.c [hunk 2]
```diff
@@ -1338,8 +1337,16 @@ native_create_actions(pe_resource_t * rsc, pe_working_set_t * data_set)
             crm_notice("See https://wiki.clusterlabs.org/wiki/FAQ#Resource_is_Too_Active for more information");
         }
 
-        if (rsc->recovery_type == recovery_stop_start) {
-            need_stop = TRUE;
+        switch (rsc->recovery_type) {
+            case recovery_stop_start:
+                need_stop = TRUE;
```
Comment (clumens) [inline lib/pacemaker/pcmk_sched_native.c:1342]: need_stop could similarly just be a bool, and in fact it's passed to schedule_restart_actions which takes a bool as an argument.

File: lib/pacemaker/pcmk_sched_native.c (hunk 1 above)
Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_native.c:1253]: Yeah, this file is high on the best-practices list, so I was going to take care of it then

**Comment by clumens** [comment]
Looks good otherwise.

**Pull request merged**